ERCS/erc-7752.md

```diff
@@ -0,0 +1,937 @@
+
+#### Identity Registry
+
+```solidity
+function identityRegistry() external view returns (IIdentityRegistry);
+function setIdentityRegistry(address identityRegistryAddress) external;
+```
+
+- **Requirements**:
+  - MUST enforce identity verification for token holders.
+  - `setIdentityRegistry` MUST only be callable by the contract owner.
+  - MUST emit the `IdentityRegistryAdded` event when updated.
+
+#### Compliance Contract
+
+```solidity
+function compliance() external view returns (IModularCompliance);
+function setCompliance(address complianceAddress) external;
+```
+
+- **Requirements**:
+  - MUST enforce regulatory compliance for token transfers.
+  - `setCompliance` MUST only be callable by the contract owner.
+  - MUST emit the `ComplianceAdded` event when updated.
+
+### Recovery Mechanisms
+
+The contract SHOULD provide mechanisms to recover tokens if a wallet is lost:
+
+```solidity
+function recoveryAddress(
+    address _lostWallet,
+    address _newWallet,
+    address _investorOnchainID,
+    uint256[] calldata _ids
+) external returns (bool);
+```
+
+- **Requirements**:
+  - MUST only be callable by an address with the `AgentRole`.
+  - MUST verify the identity of the investor through the Identity Registry.
+  - MUST emit the `RecoverySuccess` event upon successful recovery.
+
+### Locking and Unlocking Tokens
+
+The contract MUST allow token holders to lock and unlock their tokens:
+
+```solidity
+function lockTokens(uint256 tokenId, uint256 amount) external;
+function unlockTokens(uint256 tokenId, uint256 amount) external;
+```
+
+- **Requirements**:
+  - MUST prevent locked tokens from being transferred.
+  - MUST emit `TokensLocked` and `TokensUnlocked` events accordingly.
+
+### Events
+
+The contract MUST emit events as specified in the interface to enable off-chain monitoring and compliance:
+
+- `TokensLocked`
+- `TokensUnlocked`
+- `Approval`
+- `ApprovalForAll`
+- `Transfer`
+- `TransferValue`
+- `TokenInvalidated`
+- `URI`
+- `AgentAddedForToken`
+- `AgentRemovedForToken`
+- `UpdatedTokenInformation`
+- `IdentityRegistryAdded`
+- `ComplianceAdded`
+- `Paused`
+- `Unpaused`
+- `Mint`
+- `Burn`
+- `AddressFrozen`
+- `TokenFrozen`
+- `TokenUnfrozen`
+- `RecoverySuccess`
+
+### Access Control
+
+- The contract MUST implement role-based access control to restrict certain functions to authorized addresses (e.g., `AgentRole`).
+- Ownership and agent roles SHOULD be managed securely to prevent unauthorized access.
+
+### Compliance Requirements
+
+- The contract MUST integrate with compliance modules to enforce KYC/AML regulations.
+- Transfers MUST be checked against compliance rules before execution.
+
+### Error Handling
+
+- The contract SHOULD use standard OpenZeppelin error messages and revert reasons for consistency.
+- MUST revert transactions that violate compliance rules, are paused, or involve frozen addresses/tokens.
+
+### Backwards Compatibility
```

Comment:
This section is in the wrong place. It should come after the Rationale section, and be at level 2 (`## Back...`).